Cart v3

iuesver opened the pull request:
tailwind CSS로 반응형 디자인 추가
redux-persist 설치

redux-persist로 selectedItem들의 값을 localStorage에 저장하려 했지만, 새로고침시 localStorage도 초기화되는 문제 발생
이로 인해 새로고침 발생 시 장바구니 초기화

Chumsung97 committed:
- 스토어(리듀서, 액션) 추가, 스토어를 리듀서 파트에서 생성하여 메인으로 임포트
- 상점 페이지(메인, 각 카테고리별 상품 페이지, 상품 상세페이지) 병합, 라우터 추가
- 폴더 정리, 병합및 정리를 위한 코드 수정
- 메인에 슬라이드(<Index />) 추가

Chumsung97 committed:
- 스토어(리듀서, 액션) 추가, 스토어를 리듀서 파트에서 생성하여 메인으로 임포트
- 상점 페이지(메인, 각 카테고리별 상품 페이지, 상품 상세페이지) 병합, 라우터 추가
- 폴더 정리, 병합및 정리를 위한 코드 수정
- 메인에 슬라이드(<Index />) 추가

ㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡ

- src/store/store.ts 파일 리듀서 병합차 이름 변경 후 폴더이동 store.ts -> cartReducer.tsx (./assets/reducers/cartReducer.tsx)
- 리듀서 병합 완료
 ㄴ itemFilter API 원본 useSelector((state:any) => state.itemList)로 사용가능
 ㄴ reducer 장바구니용 useSelector((state:any) => state.cart)로 사용가능
- 리듀서 사용하는 컴포넌트 모두 적용 완료(ProductDescription, ProductsTable, SearchBar, Cart)

- 검색창 대소문자, 띄어쓰기 인식

- 네비게이션 -> 패션, 악세서리, 디지털, 로그인페이지 버튼 묶음 & 장바구니, 테마 버튼, searchbar 묶음(tailwind 작업 예정)
 ㄴ 아이콘 등록 필요(로그인, 테마, 장바구니)
- 메인페이지 카테고리별 상품 4개씩 진열

@제품 상세 페이지의 버튼으로 장바구니에 담는 기능 필요 - '장바구니로 이동' 버튼은 구현

iuesver committed:
1. 장바구니 반응형 CSS 추가
2. redux-persist 설치

oinochoe commented on src/components/productShow/ProductDescription.tsx:
```diff
@@ -66,7 +68,7 @@ export default function ProductDescription() {
               </ItemInfo>
-              <Shopbutton>장바구니에 담기</Shopbutton>
+              <Shopbutton onClick={() => dispatch({type : "ADD", payload : {id : selectedItem.id, count : 1}})}>장바구니에 담기</Shopbutton>
               <Shopbutton><Link to='/cart'>장바구니로 이동</Link></Shopbutton>
```
장바구니에 담는 함수를 reducer내에서 구현하는 편이 좋아 보입니다 😄

iuesver committed:
tailwindCSS 스타일 추가
BreadCrumb 컴포넌트 추가
svg 추가

iuesver committed:
네비게이션 아이콘 추가
searchBar 스타일 변경
상품 상세페이지 반응형 스타일 추가

iuesver committed:
1. 전체적인 반응형 스타일 추가
2. tailwind CSS를 통한 다크모드 구현
3. redux.state를 참고하여 navBar에서 현재 장바구니 갯수 확인기능 구현

iuesver committed:
NavBar fixed속성으로 인해 BreadCrumb 숨겨졌던 문제 해결

iuesver committed:
새로고침 시 store 리셋 문제 해결
다크모드 store 추가
누락되었던 회원가입 페이지. 다크모드 스타일 추가
검색창 문제 일부 해결 - 메인페이지 carousel 이미지 z-index 문제 발견

iuesver committed:
1. 장바구니 구매결정 팝업 추가 및 리듀서 수정
2. 일정 이하 크기의 스크린 시 햄버거 메뉴바 추가 및 구현

oinochoe reviewed (commented):
전체적으로 저번 리뷰 이후로 ui의 styled tw를 사용한 방법으로 많이 깔끔하게 정리가 잘 된 느낌이네요.

고생하셨습니다. 😄

iuesver committed:
검색창 안보이던 문제 해결

oinochoe commented on src/components/navigation/Navbar.tsx:
```diff
@@ -1,22 +1,23 @@
+import {Link, Outlet} from 'react-router-dom';
+import tw from 'tailwind-styled-components';
 import SearchBar from './SearchBar';
+import { ShoppingBagIcon, SunIcon, MoonIcon } from '@heroicons/react/outline'
```
perttier를 저장 모드로 두고 썼다면, 제일 끝에 ; 이 붙어야 할 것 같은데요.
혹시 prettier를 안쓰시고 계신지 궁금하네요 😄
